comps/dataprep/redis/multimodal_langchain/prepare_videodoc_redis.py

```diff
@@ -0,0 +1,533 @@
+
+            # Create video file name by appending identifier
+            video_name = os.path.splitext(video_file.filename)[0]
+            video_file_name = f"{video_name}_{video_id}.mp4"
+            video_dir_name = os.path.splitext(video_file_name)[0]
+
+            # Save video file in upload_directory
+            with open(os.path.join(upload_folder, video_file_name), "wb") as f:
+                shutil.copyfileobj(video_file.file, f)
+
+            # Extract temporary audio wav file from video mp4
+            audio_file = video_dir_name + ".wav"
+            print(f"Extracting {audio_file}")
+            convert_video_to_audio(
+                os.path.join(upload_folder, video_file_name), os.path.join(upload_folder, audio_file)
+            )
+            print(f"Done extracting {audio_file}")
+
+            # Load whisper model
+            print("Loading whisper model....")
+            whisper_model = load_whisper_model(model_name=WHISPER_MODEL)
+            print("Done loading whisper!")
+
+            # Extract transcript from audio
+            print("Extracting transcript from audio")
+            transcripts = extract_transcript_from_audio(whisper_model, os.path.join(upload_folder, audio_file))
+
+            # Save transcript as vtt file and delete audio file
+            vtt_file = video_dir_name + ".vtt"
+            write_vtt(transcripts, os.path.join(upload_folder, vtt_file))
+            delete_audio_file(os.path.join(upload_folder, audio_file))
+            print("Done extracting transcript.")
+
+            # Store frames and caption annotations in a new directory
+            print("Extracting frames and generating annotation")
+            extract_frames_and_annotations_from_transcripts(
+                video_id,
+                os.path.join(upload_folder, video_file_name),
+                os.path.join(upload_folder, vtt_file),
+                os.path.join(upload_folder, video_dir_name),
+            )
+            print("Done extracting frames and generating annotation")
+            # Delete temporary vtt file
+            os.remove(os.path.join(upload_folder, vtt_file))
+
+            # Ingest multimodal data into redis
+            print("Ingesting data to redis vector store")
+            ingest_multimodal(video_name, os.path.join(upload_folder, video_dir_name), embeddings)
+
+            # Delete temporary video directory containing frames and annotations
+            shutil.rmtree(os.path.join(upload_folder, video_dir_name))
+
+            print(f"Processed video {video_file.filename}")
+            end = time.time()
+            print(str(end - st))
+
+        return {"status": 200, "message": "Data preparation succeeded"}
+
+    raise HTTPException(status_code=400, detail="Must provide at least one video (.mp4) file.")
+
+
+@register_microservice(
+    name="opea_service@prepare_videodoc_redis", endpoint="/v1/dataprep/generate_captions", host="0.0.0.0", port=6007
+)
+@traceable(run_type="tool")
```

Comment:
please remove langsmith traceable

Reply:
Thank for your feedback. We have removed langsmith here.